File: tests/fixtures/ruby/user_service.rb
require 'json'
require_relative 'logger'

module Users
  class BaseService
    def name; 'base'; end
  end

  class UserService < BaseService
    def initialize(repo)
      @repo = repo
    end

    def find(id)
      @repo.find(id)
    end

    def create(attrs)
      validate(attrs)
      @repo.save(attrs)
    end

    private

    def validate(attrs)
      raise 'invalid' if attrs.nil?
    end
  end

  def self.helper(a, b)
    a + b
  end
end

File: tests/fixtures/lua/user_service.lua
local M = {}
local users = require("users")
local json = require("cjson")

function M.find(id)
  return users.get(id)
end

function M.create(name)
  local id = generate_id()
  users.save(id, name)
  return id
end

local function generate_id()
  return "x"
end

local function plain_helper(a, b)
  return a + b
end

return M
